packages/rocketchat-ui/client/lib/chatMessages.js

```diff
@@ -193,31 +193,71 @@ this.ChatMessages = class ChatMessages {
 				}
 			}
 
-			// Run to allow local encryption, and maybe other client specific actions to be run before send
-			return RocketChat.promises.run('onClientBeforeSendMessage', msgObject).then(msgObject => {
-
-				// checks for the final msgObject.msg size before actually sending the message
-				if (this.isMessageTooLong(msgObject.msg)) {
+			// checks for the final msgObject.msg size before actually sending the message
+			if (this.isMessageTooLong(msgObject.msg)) {
+				if (RocketChat.settings.get('Message_AllowAttachTooLongMessages') === true) {
+					swal({
+						text: t('Message_too_long_as_an_attachment_question'),
+						title: '',
+						type: 'warning',
+						showCancelButton: true,
+						confirmButtonText: t('Yes'),
+						cancelButtonText: t('No'),
+						closeOnConfirm: true
+					}, () => {
+						// Transform message to attachment...
+						const contentType = 'application/vnd.openxmlformats-officedocument.wordprocessingml.document';
```

Comment:
Can we have a simpler format here? Having it as `plain/text` would be way better for compatibility.

Reply:
Agreed... We need a more compatible format

Reply:
Agreed. I'll to perform it.